wolfcrypt/src/evp.c

```diff
@@ -7145,6 +7147,885 @@ void wolfSSL_EVP_PKEY_free(WOLFSSL_EVP_PKEY* key)
+    }
+    wsz = (int)XSTRLEN((const char*)buff);
+    if (wolfSSL_BIO_write(out, buff + idx, wsz) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    XSTRNCPY(line, " bit)\n", sizeof(line));
+    if (wolfSSL_BIO_write(out, line, (int)XSTRLEN(line)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    Indent(out, indent);
+    XSTRNCPY(line, "public-key:\n", sizeof(line));
+    if (wolfSSL_BIO_write(out, line, (int)XSTRLEN(line)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (PrintHexWithColon(out, publicKey, publicKeySz, indent + 4)
+                                                != WOLFSSL_SUCCESS) {
+        return WOLFSSL_FAILURE;
+    }
+    Indent(out, indent);
+    XSTRNCPY(line, "prime:\n", sizeof(line));
+    if (wolfSSL_BIO_write(out, line, (int)XSTRLEN(line)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (PrintHexWithColon(out, prime, primeSz, indent + 4) != WOLFSSL_SUCCESS) {
+        return WOLFSSL_FAILURE;
+    }
+    idx = 0;
+    XMEMSET(buff, 0, sizeof(buff));
+    Indent(out, indent);
+    XSTRNCPY(line, "generator: ", sizeof(line));
+    if (wolfSSL_BIO_write(out, line, (int)XSTRLEN(line)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (mp_set_int(&a, generator) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (mp_todecimal(&a, (char*)buff) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+    wsz = (int)XSTRLEN((const char*)buff);
+    if (wolfSSL_BIO_write(out, buff + idx, wsz) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    XSTRNCPY(line, " (0x", sizeof(line));
+    if (wolfSSL_BIO_write(out, line, (int)XSTRLEN(line)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+    idx = 0;
+    XMEMSET(buff, 0, sizeof(buff));
+    outSz = sizeof(outHex);
+    if (Base16_Encode((const byte*)&generator, 1, outHex, &outSz ) != 0) {
+        return WOLFSSL_FAILURE;
+    }
+    if (idx + 2 < (int)sizeof(buff) ) {
+        XMEMCPY(buff + idx, outHex, 2);
+        idx += 2;
+    }
+    if (wolfSSL_BIO_write(out, buff, idx) <= 0 ) {
+        return WOLFSSL_FAILURE;
+    }
+    XSTRNCPY(line, ")\n", sizeof(line));
+    if (wolfSSL_BIO_write(out, line, (int)XSTRLEN(line)) <= 0) {
+        return WOLFSSL_FAILURE;
+    }
+
+    return WOLFSSL_SUCCESS;
+}
+#endif /* WOLFSSL_DH_EXTRA */
+
+/* wolfSSL_EVP_PKEY_print_public parses the specified key then 
+ * outputs public key info in human readable format to the specified BIO.
+ * White spaces of the same number which 'indent" gives, will be added to
+ * each line to output and ignores pctx parameter.
+ * Parameters:
+ * out     bio to output dump data
+ * pkey    buffer holding public key data
+ * indent  the number of spaces for indent
+ * pctx    context(not used)
+ * Returns 1 on success, 0 or negative on error, -2 means specified key 
+ * algo is not supported.
+ * Can handle RSA, ECC, DSA and DH public keys.
+ */
+int wolfSSL_EVP_PKEY_print_public(WOLFSSL_BIO* out,
+    const WOLFSSL_EVP_PKEY* pkey, int indent, ASN1_PCTX* pctx)
+{
+    int res = 1;
```

Comment:
'res' will always be assigned a value before returning.
Don't set it to magic number 1.

Reply:
Removed the assignment.